tests/validation/tests/v3_api/test_airgap_ha.py

```diff
@@ -97,17 +111,23 @@ def test_deploy_airgap_ha_rancher(check_hostname_length):
             bastion_node.ssh_key_name, AWS_USER,
             ag_nodes[0].private_ip_address,
             public_dns, RANCHER_AG_INTERNAL_HOSTNAME,
-            RANCHER_AG_HOSTNAME))
+            REGISTRY_HOSTNAME))
     time.sleep(180)
-    setup_rancher_server()
+    setup_rancher_server(bastion_node)
 
 
-def setup_rancher_server():
+def setup_rancher_server(bastion_node):
     base_url = "https://" + RANCHER_AG_HOSTNAME
     wait_for_status_code(url=base_url + "/v3", expected_code=401)
     auth_url = base_url + "/v3-public/localproviders/local?action=login"
     wait_for_status_code(url=auth_url, expected_code=200)
-    set_url_and_password(base_url, "https://" + RANCHER_AG_INTERNAL_HOSTNAME)
+    get_bootstrap_passwd = "export KUBECONFIG=~/kube_config_config.yaml && " \
+    "/snap/bin/kubectl get secret --namespace cattle-system bootstrap-secret -o " \
+    """go-template='{{.data.bootstrapPassword|base64decode}}{{"\\n"}}'"""
+    bootstrap_passwd = bastion_node.execute_command(get_bootstrap_passwd)[0]
+    print("bootstrap password:", bootstrap_passwd.replace(" ", ""))
+    # currently can't set password in airgap setup without reworking below function
+    # set_url_and_password(base_url, "https://" + RANCHER_AG_INTERNAL_HOSTNAME)
```

Comment:
What's required to make this function work? not a big deal if we're planning to continue manually enter the password but I'm just curious.

Reply:
I'm actually not sure why it wasn't working in my initial testing. Here were my 2 attempts:
* use the extra_args setting (from earlier thread) to pass in `bootstrap_password=admin` like we do for the regular HA install. However, for whatever reason, since this method uses the `template` approach to install rancher is my hunch, this argument was not allowed and rancher install would fail everytime I tried it. So then I tried
* using the generated bootstrap password, passing it to the `set_url_and_password` function as a new argument, logging in with the bootstrap password then setting it to a proper password. However even this approach did not work and I could not figure out why. I made sure that (in testing that new function argument) the `bootstrap` password worked manually, even though the function would fail in testing with the same setting. 
 
It might be worth more investigation in the future.

Reply:
Ahhh that's a good find @slickwarren!
Do you think this might also be somewhat related to:
https://github.com/rancher/rancher/issues/37779 ?